src/renderer/dx/DxFontRenderData.cpp

```
@@ -496,26 +505,153 @@ void DxFontRenderData::SetFeatures(std::unordered_map<std::wstring_view, uint32_
+
+    return (fontStyle == DWRITE_FONT_STYLE_ITALIC) ? -12.0f :
+                                                     (fontStyle == DWRITE_FONT_STYLE_OBLIQUE) ? -20.0f :
+                                                                                                /*fontStyle == DWRITE_FONT_STYLE_NORMAL*/ 0.0f;
+}
+
+// Method Description:
+// - Converts a float fontSize into the corresponding float value to create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontSize: the old float value to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontSize
+float DxFontRenderData::DIPsToPoints(const float fontSize) noexcept
+{
+    return fontSize * (72.0f / 96.0f);
+}
+
+// Method Description:
+// - Fill any missing axis values that might be known but were unspecified, such as omitting
+//   the 'wght' axis tag but specifying the old DWRITE_FONT_WEIGHT enum
+// Arguments:
+// - fontWeight: the old DWRITE_FONT_WEIGHT enum to be converted into an axis value
+// - fontStretch: the old DWRITE_FONT_STRETCH enum to be converted into an axis value
+// - fontStyle: the old DWRITE_FONT_STYLE enum to be converted into an axis value
+// - fontSize: the number to convert into an axis value
+// - format: the IDWriteTextFormat3 to get the defined axes from
+// Return value:
+// - The fully formed axes vector
+std::vector<DWRITE_FONT_AXIS_VALUE> DxFontRenderData::GetAxisVector(const DWRITE_FONT_WEIGHT fontWeight,
+                                                                    const DWRITE_FONT_STRETCH fontStretch,
+                                                                    const DWRITE_FONT_STYLE fontStyle,
+                                                                    const float fontSize,
+                                                                    IDWriteTextFormat3* format)
+{
+    if (!format)
+    {
+        // return early
+        return {};
+    }

```

Isn't this an error that should throw instead of return an empty set?